pages/financial-info/CommonFinancialInfo.tsx

```diff
@@ -0,0 +1,218 @@
+  const [accountNumberError, setAccountNumberError] =
+    useState<string>(undefined)
+
+  const [loading, setLoading] = useState<boolean>(false)
+  const [infoMessage, setInfoMessage] = useState<string>(undefined)
+  const [errorMessage, setErrorMessage] = useState<string>(undefined)
+
+  const [branchNumber, setBranchNumber] = useState<string>(
+    financialInfo ? financialInfo.transitNumber : ''
+  )
+  const [institutionNumber, setInstitutionNumber] = useState<string>(
+    financialInfo ? financialInfo.institutionNumber : ''
+  )
+  const [accountNumber, setAccountNumber] = useState<string>(
+    financialInfo ? financialInfo.accountNumber : ''
+  )
+
+  const branch = useInternationalization('branch')
+  const institution = useInternationalization('institution')
+  const account = useInternationalization('account')
+  const submitErrorMsg = useInternationalization('submitError')
+
+  const handleReset = async (e) => {
+    e.preventDefault()
+    if (
+      confirm(
+        'You are about to reset your account details with Service Canada, are you sure you want to take this action?'
+      )
+    ) {
+      // send reset api request and set loading state
+
+      // set form state
+      setBranchNumber('')
+      setInstitutionNumber('')
+      setAccountNumber('')
+    }
+  }
+
+  const handleSubmit = async (e) => {
+    console.log('submitting...')
+    e.preventDefault()
+    e.stopPropagation()
+    try {
+      setLoading(true)
+
+      const formData = {
+        institutionNumber: institutionNumber,
+        transitNumber: branchNumber,
+        accountNumber: accountNumber,
+      }
+
+      setBranchNumberError(undefined)
+      setAccountNumberError(undefined)
+      setInstitutionNumberError(undefined)
+
+      setInfoMessage(undefined)
+      setErrorMessage(undefined)
+      console.log(userData)
+      fetch(
+        `${process.env.NEXT_PUBLIC_BASE_API_URL}/user/financial/info/${userData}`,
+        {
+          method: !financialInfo ? 'POST' : 'PUT',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(formData),
+        }
+      )
+        .then((response) => response.json())
+        .then((json) => {
+          if (json.code > 201) {
+            setErrorMessage(submitErrorMsg)
+          }
+          const messages = json.message.split(',')
+            ? json.message.split(',')
+            : json.message
+          messages.forEach((msg) => {
+            if (msg.includes('Branch')) {
+              setBranchNumberError(msg)
+            } else if (msg.includes('Account Number')) {
+              setAccountNumberError(msg)
+            } else if (msg.includes('Institution Number')) {
+              setInstitutionNumberError(msg)
+            } else {
+              setInfoMessage(msg)
+              if (json.status < 300) {
+                setFinancialInfo(json.data)
+              }
+            }
+          })
```

Comment:
Have you thought about using one `setState` call and using setState with a function to update the key you want. It would greatly update this and other components.

https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous 

This work though, so feel free to ignore!